lib/ddtrace/sampler.rb

```diff
@@ -265,7 +265,10 @@ def sample!(trace)
       trace.sampled?
     end
 
-    def_delegators :@priority_sampler, :update
+    # Forwards this method call to `@priority_sampler`, if it has a `update` method.
+    def update(rate_by_service)
+      @priority_sampler.update(rate_by_service) if @priority_sampler.respond_to?(:update)
+    end
```

Comment:
This one is a weird change that I ended up making: by setting the `post_sampler` (which is stored in the `@priority_sampler` instance variable) to an `AllSampler` instance, we get `#update` attempts in the `AllSampler`.

Because the `AllSampler` doesn't support this operation, we end up with an error.

The underlying issue is that the Ruby tracer should respect the update rate sent back by the agent, but for CiApp that is not the case. This goes against the contract we have with the Agent API: there's no clean solution here. :/